django/db/models/fields/json.py

```diff
@@ -340,12 +340,16 @@ def __init__(self, key_name, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.key_name = str(key_name)
 
-    def preprocess_lhs(self, compiler, connection):
+    def unwrap_transforms(self):
         key_transforms = [self.key_name]
         previous = self.lhs
         while isinstance(previous, KeyTransform):
             key_transforms.insert(0, previous.key_name)
             previous = previous.lhs
+        return previous, key_transforms
+
+    def preprocess_lhs(self, compiler, connection):
+        previous, key_transforms = self.unwrap_transforms()
```

Comment:
Could we make a separate commit for this change (i.e. extracting `unwrap_transforms()` from `preprocess_lhs()`) please?

Reply:
Done :+1: